src/test/java/emissary/config/ServiceConfigGuideTest.java

```diff
@@ -611,4 +611,31 @@ void testFindStringEntry_WithDefaultAndConfiguredNull_Emissary201() {
         assertEquals(nonNullValue, multiValue, "Multi-valued property should return first non-null value");
         assertEquals(defaultValue, multiNullValue, "Multi-valued property with all nulls should return default");
     }
+
+    @Test
+    void testFindRequiredStringEntry() {
+        final String workingProp = "test.workingProp";
+        final String emptyProp = "test.emptyProp";
+        final String missingProp = "test.missingProp";
+        final String nonNullValue = "non-null value";
+        final String emptyValue = "";
+
+        final ServiceConfigGuide config = new ServiceConfigGuide();
+        config.addEntry(workingProp, nonNullValue);
+        config.addEntry(emptyProp, emptyValue);
+
+        // Working entry
+        final String value = config.findRequiredStringEntry(workingProp);
+        assertEquals(value, nonNullValue, "Does this work?");
+
+        // Zero-Length Entry
+        assertThrows(IllegalArgumentException.class, () -> {
+            config.findRequiredStringEntry(emptyProp);
+        });
```

Comment:
For future reference... The `assertThrows` call actually returns the thrown Exception, so you could capture it and inspect its message.
```suggestion
        Exception ex = assertThrows(IllegalArgumentException.class, () -> config.findRequiredStringEntry(emptyProp));
        assertNotNull(ex.getMessage(), "exception message should not be null");
        assertTrue(ex.getMessage().contains(emptyProp), "Exception message should name the offending config entry");
        
```